Library/Homebrew/cmd/update-report.rb

```diff
@@ -376,16 +376,16 @@ def install_from_api_message
     api_auto_update_secs = Homebrew::EnvConfig.api_auto_update_secs.to_i
     api_auto_update_secs_default = Homebrew::EnvConfig::ENVS.fetch(:HOMEBREW_API_AUTO_UPDATE_SECS).fetch(:default)
     auto_update_secs_set = api_auto_update_secs.positive? && api_auto_update_secs != api_auto_update_secs_default
-    if !Homebrew::EnvConfig.no_auto_update? && !Homebrew::EnvConfig.no_install_from_api? && !auto_update_secs_set
+    no_auto_update_set = Homebrew::EnvConfig.no_auto_update? && !ENV["CI"]
```

Comment:
How about scoping this to `GITHUB_ACTIONS` instead? I think the change here assumes ephemeral runners and not sure that makes sense.

Reply:
You've never going to see this message unless you are already manually updating in your workflows.

Reply:
I still think it's worth notifying those people.

Reply:
I don't see how `HOMEBREW_NO_AUTO_UPDATE` + manual update is a bad workflow for CI. Particularly when it's something we do in our all of our CI that interfaces `brew install`, both ephemeral and non-ephemeral.

Reply:
> I don't see how `HOMEBREW_NO_AUTO_UPDATE` + manual update is a bad workflow for CI.

It's not that it's always bad, it's that it can be worth notifying those people in some situations and I'd rather err on the side of doing that for a message that's only output once.

> Particularly when it's something we do in our all of our CI that interfaces `brew install`, both ephemeral and non-ephemeral.

We also have some `brew doctor` opt-outs. Doesn't mean it's the right thing for everyone.